src/Symfony/Component/Intl/Resources/emoji/build.php

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env php
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+require __DIR__.'/vendor/autoload.php';
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+Builder::cleanTarget();
+$emojisCodePoints = Builder::getEmojisCodePoints();
+Builder::saveRules(Builder::buildRules($emojisCodePoints));
+
+final class Builder
+{
+    private const TARGET_DIR = __DIR__.'/../data/transliterator/emoji/';
+
+    public static function getEmojisCodePoints(): array
+    {
+        $lines = file(__DIR__.'/vendor/unicode-org/cldr/tools/cldr-code/src/main/resources/org/unicode/cldr/util/data/emoji/emoji-test.txt');
+
+        $emojisCodePoints = [];
+        foreach ($lines as $line) {
+            $line = trim($line);
+            if (!$line || str_starts_with($line, '#')) {
+                continue;
+            }
+
+            // 263A FE0F    ; fully-qualified     # ☺️ E0.6 smiling face
+            preg_match('{^(?<codePoints>[\w ]+) +; [\w-]+ +# (?<emoji>.+) E\d+\.\d+ ?(?<name>.+)$}Uu', $line, $matches);
+            if (!$matches) {
+                throw new \DomainException("Could not parse line: \"$line\".");
+            }
+
+            $codePoints = strtolower(trim($matches['codePoints']));
+            $emojisCodePoints[$codePoints] = $matches['emoji'];
+            // We also add a version without the "Zero Width Joiner"
+            $codePoints = str_replace('200d ', '', $codePoints);
+            $emojisCodePoints[$codePoints] = $matches['emoji'];
+        }
+
+        return $emojisCodePoints;
+    }
+
+    public static function buildRules(array $emojisCodePoints): Generator
```

Comment:
`: iterable` ?

Reply:
It's internal tooling. Does it change/improve something?

Reply:
It hides an implementation detail when callers only need to iterate.
Not sure it improves things though, especially in an internal API.